x-pack/plugins/lens/public/drag_drop/drag_drop.test.tsx

```diff
@@ -194,4 +193,171 @@ describe('DragDrop', () => {
+            <DragDrop
+              label="1"
+              draggable
+              droppable
+              dragType="reorder"
+              dropType="reorder"
+              itemsInGroup={['1']}
+              value={{ id: '1' }}
+              onDrop={jest.fn()}
+              dropTo={jest.fn()}
+            >
+              <div />
+            </DragDrop>
+          </ReorderProvider>
+        </ChildDragDropProvider>
+      );
+      expect(component.find(ReorderableDragDrop)).toHaveLength(0);
+    });
+    test(`Reorderable component renders properly`, () => {
+      const component = mountComponent(undefined, jest.fn());
+      expect(component.find(ReorderableDragDrop)).toHaveLength(3);
+    });
+    test(`Elements between dragged and drop get extra class to show the reorder effect when dragging`, () => {
+      const component = mountComponent({ id: '1' }, jest.fn());
+      const dataTransfer = {
+        setData: jest.fn(),
+        getData: jest.fn(),
+      };
+      component
+        .find(ReorderableDragDrop)
+        .first()
+        .find('[data-test-subj="lnsDragDrop"]')
+        .simulate('dragstart', { dataTransfer });
+      jest.runAllTimers();
+
+      component.find('[data-test-subj="lnsDragDrop-reorderableDrop"]').at(2).simulate('dragover');
+      expect(component.find('[data-test-subj="lnsDragDrop"]').at(0).prop('style')).toEqual({});
+      expect(component.find('[data-test-subj="lnsDragDrop"]').at(1).prop('style')).toEqual({
+        transform: 'translateY(-4px)',
+      });
+      expect(component.find('[data-test-subj="lnsDragDrop"]').at(2).prop('style')).toEqual({
+        transform: 'translateY(-4px)',
+      });
+
+      component.find('[data-test-subj="lnsDragDrop-reorderableDrop"]').at(2).simulate('dragleave');
+      expect(component.find('[data-test-subj="lnsDragDrop"]').at(1).prop('style')).toEqual({});
+      expect(component.find('[data-test-subj="lnsDragDrop"]').at(2).prop('style')).toEqual({});
+    });
+    test(`Dropping an item runs onDrop function`, () => {
+      const preventDefault = jest.fn();
+      const stopPropagation = jest.fn();
+      const onDrop = jest.fn();
+
+      const component = mountComponent({ id: '1' }, onDrop);
+
+      component
+        .find('[data-test-subj="lnsDragDrop-reorderableDrop"]')
+        .at(1)
+        .simulate('drop', { preventDefault, stopPropagation });
+      expect(preventDefault).toBeCalled();
+      expect(stopPropagation).toBeCalled();
+      expect(onDrop).toBeCalledWith({ id: '1' });
+    });
+    test(`Keyboard navigation: user can reorder an element`, () => {
+      const onDrop = jest.fn();
+      const dropTo = jest.fn();
+      const component = mountComponent({ id: '1' }, onDrop, dropTo);
+      const keyboardHandler = component
+        .find(ReorderableDragDrop)
+        .at(1)
+        .find('[data-test-subj="lnsDragDrop-keyboardHandler"]');
+
+      keyboardHandler.simulate('keydown', { key: 'Enter' });
```

Comment:
This is testing using the Enter key but the aria labels say to use spacebar. Both seem to work, but there's inconsistency?

Reply:
Changed it to space.